service/src/main/java/bio/terra/externalcreds/services/ProviderService.java

```diff
@@ -39,7 +39,8 @@
 @Slf4j
 public class ProviderService {
   public final ExternalCredsConfig externalCredsConfig;
-  public final ProviderClientCache providerClientCache;
+  public final ProviderOAuthClientCache providerOAuthClientCache;
+  public final ProviderTokenClientCache providerTokenClientCache;
```

Comment:
It doesn't look like `ProviderService` uses the `providerTokenClientCache`, so we can just have it be a member if the `TokenProviderService` subclass